src/Uno.Extensions.Localization.UI/LocalizationService.cs

```diff
@@ -27,17 +27,29 @@ public CultureInfo CurrentCulture
 			var settingsCulture = _settings?.CurrentValue?.CurrentCulture?.AsCulture();
 			if (settingsCulture is null)
 			{
-				var defaultCulture = PrimaryLanguageOverride ??
-							CultureInfo.DefaultThreadCurrentUICulture?.Name ??
-							CultureInfo.DefaultThreadCurrentCulture?.Name ??
-							_uiThread?.CurrentUICulture?.Name ??
-							_uiThread?.CurrentCulture?.Name;
-				settingsCulture = string.IsNullOrWhiteSpace(defaultCulture) ?
+				var defaultLanguage = PrimaryLanguageOverride ?? string.Empty;
+
+				var defaultCulture =
+							CultureInfo.DefaultThreadCurrentUICulture ??
+							CultureInfo.DefaultThreadCurrentCulture ??
+							_uiThread?.CurrentUICulture ??
```

Comment:
```suggestion
							CultureInfo.CurrentUICulture ??
```

As per [the docs](https://learn.microsoft.com/dotnet/api/system.threading.thread.currentuiculture?view=net-7.0#remarks):

> The [CurrentUICulture](https://learn.microsoft.com/en-us/dotnet/api/system.threading.thread.currentuiculture?view=net-7.0#system-threading-thread-currentuiculture) property doesn't work reliably when used with any thread other than the current thread. In .NET Framework, reading the property is reliable, although setting it for a thread other than the current thread is not. On .NET Core, an [InvalidOperationException](https://learn.microsoft.com/en-us/dotnet/api/system.invalidoperationexception?view=net-7.0) is thrown if a thread attempts to read or write the [CurrentUICulture](https://learn.microsoft.com/en-us/dotnet/api/system.threading.thread.currentuiculture?view=net-7.0#system-threading-thread-currentuiculture) property on a different thread. **We recommend that you use the [CultureInfo.CurrentUICulture](https://learn.microsoft.com/en-us/dotnet/api/system.globalization.cultureinfo.currentuiculture?view=net-7.0#system-globalization-cultureinfo-currentuiculture) property to retrieve and set the current culture.**

I would question whether it's desirable to read the culture of a specific thread rather than what the docs say. What do you think @nickrandolph ?

Reply:
Good catch. Let's address in different PR. This PR is predominantly to change the order things are done in.